### ISSUE TITLE
Axios 0.21.1 and 0.20.0 not merging url with baseURL

### ISSUE THREAD

**Issue by sebnapi** [axios/axios#3663]
#### Describe the bug
I was just encountering the problem locally, that the new axios version is not merging `url` with `baseURL`. I found some issues with version 0.19.0 Issue #2190 where the params are not merged. I used the runnable example to showcase my problem:

#### To Reproduce
https://runkit.com/sebnapi/axios-0-21-1

```js
const axios_newer = require("axios@0.21.1");

const ctr_args = {
    baseURL: "https://httpbin.org",
    url: "/anything",
    params: {
      q: "question",
    }
  };

axios_newer.request().then(function(r) { console.log("axios 0.21", r.data); }).catch(function (e) {console.log(e)})
```

#### Expected behavior
Request gets fullfilled with uri appended to baseURL.

#### Environment
 - Axios Version 0.21.0, 0.20.0, 0.19.0

#### Additional context/Screenshots

![Screenshot 2021-03-03 at 14 48 50](https://user-images.githubusercontent.com/539707/109823196-9e88ba80-7c2f-11eb-8bf9-7007e87910de.png)

**Comment by chinesedfan** [comment]
See https://github.com/axios/axios/issues/3579#issuecomment-770341715.

**Comment by amaredeus** [comment]
Yep I'm having the same issue although slightly different use case

CommonAPI.js
```javascript
const axios = require('axios');

const someAPI = axios.create({
    baseUrl: 'https://someapiwebsite/v1/',
    timeout: 1000,
    headers: {
        'Some-Api-Version': '1.0.0',
        'Authorization': process.env.MY_TOKEN
    }
});

module.exports = someAPI;
```

Calling statement
```javascript
const someAPI = require('../../somehwhere/SomeAPI.js');
const blocks = someAPI.get(`blocks/${FAQ_PAGE_ID}/children`);
```

**Comment by laygir** [comment]
Yup exactly. I'm also having this issue as I just updated my axios version in the project.

I cloned runkit from @sebnapi notice on version 0.21.1 the `config.url` value is not merged with baseURL.
I'm expecting same behaviour as older versions.

https://runkit.com/liger/60eedbdbb3e7b0001a2dc521

I was able to trace back to this PR: https://github.com/axios/axios/pull/2391
However this PR should have fixed the relative baseURL issue but somehow absolute `baseURL` also is not prepended to the `url`
